core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportClusterAllocationExplainAction.java

```diff
@@ -161,6 +166,9 @@ public static NodeExplanation calculateNodeExplanation(ShardRouting shard,
         if (node.getId().equals(assignedNodeId)) {
             finalDecision = ClusterAllocationExplanation.FinalDecision.ALREADY_ASSIGNED;
             finalExplanation = "the shard is already assigned to this node";
+        } else if (shard.unassigned() && hasPendingAsyncFetch) {
```

Comment:
When I look at the `allocateUnassigned` method for both the `PrimaryShardAllocator`(0) and `ReplicaShardAllocator`(1), it checks if data is still pending and skips the allocation for either before doing any sort of decider check it looks like though?

(0): https://github.com/elastic/elasticsearch/blob/ee4e470f6000fda030ad1f6d9216efec70119227/core/src/main/java/org/elasticsearch/gateway/PrimaryShardAllocator.java#L105-L111
(1): https://github.com/elastic/elasticsearch/blob/ee4e470f6000fda030ad1f6d9216efec70119227/core/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java#L82-L86